### ISSUE TITLE
StructureTree is not available anymore on PDFDocumentProxy, how to get it?

### ISSUE THREAD

**Issue by BennyAlex** [mozilla/pdf.js#16352]
I am relying on the internal prop StructureTree of PDFDocumentProxy, but unfortunatelly it was removed in newer versions of pdf.js.

Can you add it back or provide another way of accessing this?

Thanks, have a good day.

**Comment by Snuffleupagus** [comment]
> I am relying on the internal prop StructureTree of PDFDocumentProxy, but unfortunatelly it was removed in newer versions of pdf.js.

Please note that we've never exposed any StructTree-data on `PDFDocumentProxy`, since that's *page-specific* data; refer to the following `PDFPageProxy` method:
https://github.com/mozilla/pdf.js/blob/598408b7af9688e4a85d47fe47227ce32f993017/src/display/api.js#L1665-L1672

*For future reference:* When opening an issue, please provide *all* of the information requested in https://github.com/mozilla/pdf.js/blob/master/.github/ISSUE_TEMPLATE.md since as-is this issue is unfortunately missing too much information to be actionable/valid.

**Comment by BennyAlex** [comment]
I (or more exactly the verapdf-js-libary) is using this call 
const onDocumentLoadSuccess = useCallback(
    async (data) => {
      console.log('onDocumentLoadSuccess', data);
      setStructureTree(data._pdfInfo.structureTree);
      setLoaded(true);
    },
    [props.onLoadSuccess, bboxes]
  );

this is using the react-pdf wrapper. The data had this structTree prop:
See here:
https://github.com/veraPDF/verapdf-js-viewer/blob/d21d16722a08872cac9b13aefdbf23f7266bf917/src/components/pdfDocument/PdfDocument.tsx#L196

This is comming from the document directly, not any specific page?

**Comment by BennyAlex** [comment]
@Snuffleupagus No they use pdf.js under the hood, and this is the DocumentProxy Object comming directly from pdf.js.

I also ask them, they said its comming from here.

**Comment by Snuffleupagus** [comment]
> and this is the DocumentProxy Object comming directly from pdf.js.

No, they clearly have *modified* it to contain additional data that we've never provided in that way. Besides, https://github.com/mozilla/pdf.js/issues/16352#issuecomment-1523443617 already show you how to access the StructTree-data.

**Comment by Snuffleupagus** [comment]
By your own account in https://github.com/mozilla/pdf.js/issues/16352#issuecomment-1523807883 you're apparently using the following project: https://github.com/veraPDF/verapdf-js-viewer/

[Based on this code](https://github.com/veraPDF/verapdf-js-viewer/blob/d21d16722a08872cac9b13aefdbf23f7266bf917/src/components/pdfDocument/PdfDocument.tsx#L29-L33) this is functionality added by that project, which we simply cannot support here, hence please address further questions to https://github.com/veraPDF/verapdf-js-viewer/

**Comment by BennyAlex** [comment]
this is simply a type definition. Nowhere in the code a structTree is beeing set

**Comment by Snuffleupagus** [comment]
> I cant find where the structuredTree is beeing set inside vera-pdf (or in react-pdf).

We know nothing about either of those projects here, pretty please just direct further questions on this topic to https://github.com/veraPDF/verapdf-js-viewer/

**Comment by BennyAlex** [comment]
As you can See here
https://github.com/veraPDF/verapdf-js-viewer/blob/d21d16722a08872cac9b13aefdbf23f7266bf917/package.json#LL46

They use V2 of pdf.js. How can you be sure that the prob Was not there?

**Comment by Snuffleupagus** [comment]
> As you can See here https://github.com/veraPDF/verapdf-js-viewer/blob/d21d16722a08872cac9b13aefdbf23f7266bf917/package.json#LL46

But that's their own *fork* of the official pdfjs-dist repository, which may thus contain any number of changes and modifications from the original PDF.js library; we really cannot help you with modifications done by a third-party.

*Once more:* Please address your questions to https://github.com/veraPDF/verapdf-js-viewer

**Comment by BennyAlex** [comment]
you can look at their forks, nothing changed. So its this projects changes...
